controllers/ratings.js

```diff
@@ -0,0 +1,90 @@
+import models from '../models';
+
+const { articles, ratings, users } = models;
+class Rating {
+  /**
+   * @param {object} req Request sent to the route
+   * @param {object} res Response from server
+   * @returns {object} Object representing the response returned
+   */
+  static async rateArticle(req, res) {
+    try {
+      const { articleSlug } = req.params;
+      const { rating } = req.body;
+
+      const findArticle = await articles.findOne({
+        where: { slug: articleSlug },
+        attributes: ['id', 'title', 'slug', 'body'],
+        include: [{ model: users, as: 'author' }]
+      });
+
+      if (findArticle === null) {
+        return res.status(404).json({ error: 'failed to find an article to rate' });
+      }
+
+      const {
+        title, body, slug, author
+      } = findArticle;
+
+
+      const user = await users.findOne({ where: { email: req.decoded.email } });
+
+      const findRate = await ratings.findOne({
+        where: { articleSlug: slug, userId: user.id }, attributes: ['rating']
+      });
+
+      switch (true) {
+        case findRate === null:
```

Comment:
in which cases are you using the `switch` statement
```javascript
if (!findRate) {
  await ratings.create({
    rating,
    articleSlug: slug,
    userId: user.id
  });
  return res.status(201).json({
    message: 'you have successfully rated this article',
    rating: {
      rating,
      author: author.username,
      title,
      body
    }
  });
}
return res.status(200).json({
  message: 'you already rated this article',
  rating
});
```
may my request be put under your consideration. But kindly compare and check before you change